integration information and database guides (#6986)
cloudflare/cloudflare-docs: content/workers/learning/integrations/_index.md

---
pcx_content_type: concept
title: Integrations
weight: 6
---

# Integrations

{{<directory-listing>}}

One of the key features of Cloudflare Workers is the ability to easily integrate with other services and products. In this document, we will explain the types of integrations available with Cloudflare Workers and provide step-by-step instructions for using them.

## Types of Integrations
Cloudflare Workers offers several types of integrations, including:

> * [Databases](/workers/platform/integrations/databases/): Cloudflare Workers can be easily integrated with a variety of databases, including SQL and NoSQL databases. This allows you to store and retrieve data from your databases directly from your Cloudflare Workers code.

Reviewer: ```suggestion
* [Databases](/workers/platform/integrations/databases/): Cloudflare Workers can be integrated with a variety of databases, including SQL and NoSQL databases. This allows you to store and retrieve data from your databases directly from your Cloudflare Workers code.
```

* [APIs](/workers/platform/integrations/apis/): Cloudflare Workers can be used to integrate with external APIs, allowing you to access and use the data and functionality exposed by those APIs in your own code.
* [Third-party services](/workers/platform/integrations/external-services/): Cloudflare Workers can be used to integrate with a wide range of third-party services, such as payment gateways, authentication providers, and more. This makes it possible to use these services in your Cloudflare Workers code.

> ## How to Use Integrations

Author: I know of one example, but certainly we could use a few more.

> ## How to Use Integrations

Author: I'm going to put some more examples as an action item for the future.

Using integrations with Cloudflare Workers is easy. Here is a general step-by-step guide for using any of the available integrations:

* Determine which integration you want to use and make sure you have the necessary accounts and credentials for it.
* In your Cloudflare Workers code, import the necessary libraries or modules for the integration.
* Use the provided APIs and functions to connect to the integration and access its data or functionality.
* Store necessary secrets and keys using secrets via [`wrangler secret put <KEY>`](/workers/wrangler/commands/#secret)

## Tips and Best Practices
Here are a few tips and best practices to help you get the most out of using integrations with Cloudflare Workers:

> * Be sure to properly secure your integrations and protect sensitive data. Ensure you use secure authentication and authorization where possible, and ensure the validity of libraries you import.

Reviewer 2: ```suggestion
* Secure your integrations and protect sensitive data. Ensure you use secure authentication and authorization where possible, and ensure the validity of libraries you import.
```

> * Use caching to improve performance and reduce the load on an external service.

Author: Added a link!

* Split your Workers into service-oriented architecture using [Service bindings](/workers/platform/bindings/about-service-bindings/) to make your application more modular, easier to maintain, and more performant.

> * We recommend using [Custom Domains](/workers/platform/triggers/custom-domains/) when communicating with external APIs and services, which create a DNS record on your behalf and treat your Worker as an application instead of a proxy.

Reviewer 2: ```suggestion
* Use [Custom Domains](/workers/platform/triggers/custom-domains/) when communicating with external APIs and services, which create a DNS record on your behalf and treat your Worker as an application instead of a proxy.
```
